flutter_src/lib/logic/api/firestore_db.dart

```diff
@@ -1,32 +1,58 @@
 import 'package:cloud_firestore/cloud_firestore.dart';
 import 'package:macrobaseapp/logic/state/macro_state/macro_notifier.dart';
+import 'package:macrobaseapp/logic/state/team_state/team_notifier.dart';
 import 'package:macrobaseapp/model/adapters/macro_model.dart';
+import 'package:macrobaseapp/model/adapters/team_model.dart';
 import 'package:macrobaseapp/model/entities/macro.dart';
+import 'package:macrobaseapp/model/entities/team.dart';
 
 class FirestoreService {
   final Firestore db;
 
   FirestoreService({FirestoreService db})
     : db = db ?? Firestore.instance;
 
-  void getMacros(MacroNotifier macroNotifier, String creatorId) async {
-    QuerySnapshot snapshot = await db.collection('macros').where("creatorId", isEqualTo: creatorId).getDocuments();
+  void getEntries(TeamNotifier teamNotifier, MacroNotifier macroNotifier, String creatorId) {
+    _getMacros(macroNotifier, creatorId);
+    _getTeams(teamNotifier, creatorId);
+  }
+
+  void _getMacros(MacroNotifier macroNotifier, String creatorId) async {
+    QuerySnapshot snapshot = await db.collection('macros').where(
+        "creatorId", isEqualTo: creatorId).getDocuments();
 
     List<Macro> _macroList = [];
 
     snapshot.documents.forEach((element) {
-      try{
+      try {
         Macro macro = MacroModel.fromJson(element.data);
         macro.macroId = element.documentID;
         _macroList.add(macro);
-      } catch(e) {
+      } catch (e) {
         print(element.data);
       }
     });
 
     macroNotifier.macroList = _macroList;
   }
 
+  void _getTeams(TeamNotifier teamNotifier, String creatorId) async {
+    QuerySnapshot snapshot = await db.collection('teams').where("creatorId", isEqualTo: creatorId).getDocuments();
+
+    List<Team> _teamList = [];
+
+    snapshot.documents.forEach((element) {
+      try{
+        Team team = TeamModel.fromJson(element.data);
+        team.teamId = element.documentID;
+        _teamList.add(team);
+      } catch(e) {
+        print(element.data);
+      }
+    });
+
+    teamNotifier.teamList = _teamList;
+  }
```

Comment:
These two functions look really similar, however, I wasn't sure how to combine them into one generic function.

Reply:
yeah, I can see the structural similarity, but the types make them fairly different, so I think they are better to be kept separated. BTW, you fixed the style in the top onem but not the bottom one.

Reply:
Just FYI: In Java you would use generics to do it, but again, not worth the effort here... too simple
https://docs.oracle.com/javase/tutorial/extra/generics/methods.html

Reply:
Thanks for letting me know about Generics! Yea, in this case it seems okay to keep it this way for now.